mypy/semanal.py

```diff
@@ -2748,6 +2748,13 @@ def analyze_name_lvalue(self,
         existing = names.get(name)
 
         outer = self.is_global_or_nonlocal(name)
+        if not outer and isinstance(self.type, TypeInfo) and self.type.is_enum:
```

Comment:
Also require that `kind == MDEF`? Test that having a local variable the name of which overlaps with an enum member name in a method defined within an enum class is okay.

Reply:
Sure!

Reply:
Yes, there was a bug here. Thanks for the suggestion! 👍